test/jdk/com/sun/net/httpserver/HttpsParametersClientAuthTest.java

```diff
@@ -0,0 +1,379 @@
+import javax.net.ssl.SSLParameters;
+import javax.net.ssl.TrustManagerFactory;
+
+import com.sun.net.httpserver.HttpExchange;
+import com.sun.net.httpserver.HttpHandler;
+import com.sun.net.httpserver.HttpsConfigurator;
+import com.sun.net.httpserver.HttpsParameters;
+import com.sun.net.httpserver.HttpsServer;
+import jdk.test.lib.net.SimpleSSLContext;
+import jdk.test.lib.net.URIBuilder;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.params.ParameterizedTest;
+import org.junit.jupiter.params.provider.ValueSource;
+import static java.net.http.HttpClient.Builder.NO_PROXY;
+import static java.net.http.HttpClient.Version.HTTP_1_1;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertFalse;
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.fail;
+
+/*
+ * @test
+ * @bug 8326381
+ * @summary verifies that the setNeedClientAuth() and setWantClientAuth()
+ *          methods on HttpsParameters class work as expected
+ * @library /test/lib
+ * @build jdk.test.lib.net.SimpleSSLContext jdk.test.lib.net.URIBuilder
+ * @run junit HttpsParametersClientAuthTest
+ */
+public class HttpsParametersClientAuthTest {
+
+    private static final AtomicInteger TID = new AtomicInteger();
+    private static final ThreadFactory SRV_THREAD_FACTORY = (r) -> {
+        final Thread t = new Thread(r);
+        t.setDaemon(true);
+        t.setName("server-thread-" + TID.incrementAndGet());
+        return t;
+    };
+
+    /**
+     * verifies default values of {@link HttpsParameters#setNeedClientAuth(boolean)}
+     * and {@link HttpsParameters#setWantClientAuth(boolean)} methods
+     */
+    @Test
+    public void testDefaultClientAuth() throws Exception {
+        // test default values
+        HttpsParameters defaultParams = new Params();
+        assertFalse(defaultParams.getNeedClientAuth(),
+                "needClientAuth was expected to be false but wasn't");
+        assertFalse(defaultParams.getWantClientAuth(),
+                "wantClientAuth was expected to be false but wasn't");
+    }
+
+    /**
+     * sets {@link HttpsParameters#setNeedClientAuth(boolean)} to true and verifies
+     * that subsequent calls to {@link HttpsParameters#getNeedClientAuth()} returns
+     * true and {@link HttpsParameters#getWantClientAuth()} returns false
+     */
+    @ParameterizedTest
+    @ValueSource(booleans = {true, false})
+    public void testNeedClientAuth(final boolean initialWantClientAuth) throws Exception {
+        HttpsParameters needClientAuthParams = new Params();
+        // first set wantClientAuth to an initial value to verify that it later gets reset
+        needClientAuthParams.setWantClientAuth(initialWantClientAuth);
+        // needClientAuth = true and thus wantClientAuth = false
+        needClientAuthParams.setNeedClientAuth(true);
```

Comment:
I just want to clarify that these new cases look not suggested by my previous comment.

In my suggested cases, as shown as the below, 
```
setNeedClientAuth(true);
setWantClientAuth(false);
```
```
setWantClientAuth(true);
setNeedClientAuth(false);
```
this first setting takes a state (`needClientAuth` or `wantClientAuth`) to `true`, and the second setting takes both states to `false`.
Before this fix, the second setting doesn't reset the `true` state specified by the first setting to `false`.

Of course, if you think it's unnecessary to cover these corner cases, please feel free to ignore them.

Reply:
Hello John, it wasn't clear to me previously that this is the specific test you were suggesting. I have now updated the PR to introduce this additional test.